glib-macros/tests/props.rs

```diff
@@ -0,0 +1,57 @@
+use glib::prelude::*;
+#[test]
+fn props() {
+    mod foo {
+        use glib::prelude::*;
+        use glib::subclass::prelude::*;
+        use glib_macros::Props;
+        use std::cell::RefCell;
+
+        mod imp {
+            use super::*;
+
+            #[derive(Props, Default)]
+            pub struct Foo {
+                #[prop("bar", "bar", "This is a bar", None, glib::ParamFlags::READWRITE)]
+                bar: RefCell<String>,
+                #[prop("buzz", "buzz", "This is a buzz", 1, 100, 1, glib::ParamFlags::READWRITE)]
+                buzz: RefCell<u32>,
+            }
+
+            #[glib::object_subclass]
+            impl ObjectSubclass for Foo {
+                const NAME: &'static str = "MyFoo";
+                type Type = super::Foo;
+            }
+        }
+
+        pub trait FooExt: 'static {
+            fn test(&self);
+        }
+
+        impl<O: IsA<Foo>> FooExt for O {
+            fn test(&self) {
+                let _self = self.as_ref().downcast_ref::<Foo>().unwrap().imp();
+                unimplemented!()
+            }
+        }
+
+        glib::wrapper! {
+            pub struct Foo(ObjectSubclass<imp::Foo>);
+        }
+    }
+
+    let myfoo: foo::Foo = glib::object::Object::new(&[]).unwrap();
+
+    let bar: String = myfoo.property("bar");
```

Comment:
Something that would be nice to have is to auto generate the getter/setter/connect_notify by default

Reply:
There should be some way to override the getter/setter at least, since you may want to implement custom behavior.

Reply:
Yes, see my comment above.

Reply:
Ok, I'll do it

Reply:
After searching a bit, I think this is impossible. I would have to make two derive macros communicate, but that's not possible:
```rust
mod imp {
  #[derive(Props)]
  struct Foo{
    #[prop(get, set)]
    name: RefCell<String>
  }
}

// Not possible, I can't get the data from `Props` macro
#[derive(PropsAccess)]
struct Foo{}

// Best we can do
#[derive(PropsAccess)]
#[props[(name: String), (prop_name: PropType), ...]]
struct Foo{}
```

Reply:
getters/setters are mostly done, currently implemented on a `{Type}Ext` trait.
Should I also generate `connect_notify_{param}` for each param? I prefer not.